src/xds/client.rs

```diff
@@ -181,86 +243,142 @@ impl AdsClient {
         };
 
         info!("Starting stream");
-        let mut _response_stream = client
+        let mut respones_stream = client
             .delta_aggregated_resources(tonic::Request::new(outbound))
             .await
             .map_err(Error::Connection)?
             .into_inner();
 
         info!("Stream established");
 
-        while let Some(response) = _response_stream.message().await? {
-            info!(
-                type_url = response.type_url,
-                size = response.resources.len(),
-                "received response"
-            );
-            let mut updates: Vec<XdsUpdate<Workload>> = Vec::new();
-            // TODO: on reconnect we need to handle this as SotW I imagine?
-            for res in response.resources.iter().cloned() {
-                let resource = Resource::try_from(res).unwrap();
-                match resource {
-                    Resource::Workload(inner) => {
-                        let name = inner.name.clone();
-                        let up = XdsUpdate::Update(inner);
-                        updates.push(up);
-                        workloads.insert(name);
-                    }
+        let demand = &mut self.demand;
+        loop {
+            tokio::select! {
+                demand_event = demand.recv() => {
```

Comment:
Can we break up `run_internal`? It's kind of hard to read. 

* outbound_stream(watches vec<&str>, discovery_req_rx, Receiver<DiscoveryRequest>) -> Stream
* One for each of these select blocks.